tests/components/nest/test_config_flow_sdm.py

```diff
@@ -1,74 +1,203 @@
+        domain=DOMAIN,
+        data={
+            "auth_implementation": DOMAIN,
+            "token": {
+                # Verify this is replaced at end of the test
+                "access_token": "some-revoked-token",
             },
-            "http": {"base_url": "https://example.com"},
+            "sdm": {},
         },
     )
+    old_entry.add_to_hass(hass)
+
+    entry = get_config_entry(hass)
+    assert entry.data["token"] == {
+        "access_token": "some-revoked-token",
+    }
+
+    await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_REAUTH}, data=old_entry.data
+    )
+
+    # Advance through the reauth flow
+    flows = hass.config_entries.flow.async_progress()
+    assert len(flows) == 1
+    assert flows[0]["step_id"] == "reauth_confirm"
+
+    # Run the oauth flow
+    result = await hass.config_entries.flow.async_configure(flows[0]["flow_id"], {})
+    await oauth.async_oauth_flow(result)
+
+    # Verify existing tokens are replaced
+    entry = get_config_entry(hass)
+    entry.data["token"].pop("expires_at")
+    assert entry.data["token"] == {
+        "refresh_token": "mock-refresh-token",
+        "access_token": "mock-access-token",
+        "type": "Bearer",
+        "expires_in": 60,
+    }
+
+
+async def test_single_config_entry(hass):
+    """Test that only a single config entry is allowed."""
+    old_entry = MockConfigEntry(
+        domain=DOMAIN, data={"auth_implementation": DOMAIN, "sdm": {}}
+    )
+    old_entry.add_to_hass(hass)
+
+    assert await setup.async_setup_component(hass, DOMAIN, CONFIG)
 
     result = await hass.config_entries.flow.async_init(
         DOMAIN, context={"source": config_entries.SOURCE_USER}
     )
-    state = config_entry_oauth2_flow._encode_jwt(
-        hass,
-        {
-            "flow_id": result["flow_id"],
-            "redirect_uri": "https://example.com/auth/external/callback",
-        },
+    assert result["type"] == "abort"
+    assert result["reason"] == "single_instance_allowed"
+
+
+async def test_unexpected_existing_config_entries(hass, oauth):
+    """Test Nest reauthentication with multiple existing config entries."""
+
+    assert await setup.async_setup_component(hass, DOMAIN, CONFIG)
+
+    old_entry = MockConfigEntry(
+        domain=DOMAIN, data={"auth_implementation": DOMAIN, "sdm": {}}
     )
+    old_entry.add_to_hass(hass)
 
-    oauth_authorize = OAUTH2_AUTHORIZE.format(project_id=PROJECT_ID)
-    assert result["url"] == (
-        f"{oauth_authorize}?response_type=code&client_id={CLIENT_ID}"
-        "&redirect_uri=https://example.com/auth/external/callback"
-        f"&state={state}&scope=https://www.googleapis.com/auth/sdm.service"
-        "+https://www.googleapis.com/auth/pubsub"
-        "&access_type=offline&prompt=consent"
+    old_entry = MockConfigEntry(
+        domain=DOMAIN, data={"auth_implementation": DOMAIN, "sdm": {}}
     )
+    old_entry.add_to_hass(hass)
```

Comment:
This will never happen.

Reply:
Will revist this pending other comment.

Reply:
I added a comment to the test to explain the scenario of mutliple old old config entry in more detail since this is no longer a new state that can be added.